### PR TITLE
Implement requireJS and requireCSS

### PR DESCRIPTION
**Pull request by DarkSide666**
* implements requireJS and requireCSS methods for App
* adds documentation of new metods
* App now have new property $defaultTemplate = 'html.html' for better configurability

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -294,6 +303,34 @@ public function url($args = [])
...
     /**
+     * Adds additional JS script include in aplication template.
+     *
+     * @param string $url
+     *
+     * @return $this
+     */
+    public function requireJS($url)
+    {
+        $this->html->template->appendHTML('HEAD', '<script src="'.$this->url($url, null).'"></script>');
```
Comment (romaninsh) [inline src/App.php]: this should't call url, should simply use the $url , for simplicity.

File: src/App.php [hunk 2]
```diff
@@ -282,7 +291,7 @@ public function url($args = [])
         $page = $args[0];
         unset($args[0]);
 
-        $url = $page ? ($page.'.php') : '';
+        $url = $page ? ($page.($extension ? '.'.$extension : '')) : '';
```
Comment (romaninsh) [inline src/App.php]: i want to encourage integrators (people who integrate Agile UI into a framework) to rewrite url() method with the goal to properly create links between PAGES. Passing extension to a page is not right.

Comment (romaninsh) [inline src/App.php]: I think we should rename url() into page() in the future. The full name could be getPageURL().

File: src/App.php [hunk 3]
```diff
@@ -157,7 +165,7 @@ public function initLayout($layout, $options = [])
         }
         $layout->app = $this;
 
-        $this->html = new View(['defaultTemplate'=>'html.html']);
+        $this->html = new View(['defaultTemplate' => $this->defaultTemplate]);
```
Comment (romaninsh) [inline src/App.php]: application doesn't really have a template, as it's not extending View. Having defaultTemplate would be confusing here. If possible leave it 'html.html' for now as I don't want people to tamper with this too much at this point.

File: docs/app.rst [hunk 4]
```diff
@@ -40,13 +40,25 @@ this method must respond with a properly formatted url such as::
...
+Includes
+========
+
+.. php:method:: requireJS($url)
+
+Method to include additional JavaScript file in page.
+
+.. php:method:: requireCSS($url)
+
+Method to include additional CSS stylesheet in page.
```
Comment (romaninsh) [inline docs/app.rst]: example would be nice here. Advise people to use CDN.

File: src/App.php (hunk 3 above)
Comment (romaninsh) [inline src/App.php]: at best we can make it possible for someone to inject "html" property, so 

if (!$this->html) {
 ...
}

**Comment by DarkSide666** [comment]
all change requests addressed

File: src/App.php [hunk 5]
```diff
@@ -311,7 +305,7 @@ public function url($args = [], $extension = 'php')
      */
     public function requireJS($url)
     {
-        $this->html->template->appendHTML('HEAD', '<script src="'.$this->url($url, null).'"></script>');
+        $this->html->template->appendHTML('HEAD', '<script src="'.$url.'"></script>');
```
Comment (romaninsh) [inline src/App.php]: wonder if we can use getTag here...

**Comment by romaninsh** [comment]
implements #50

**Pull request merged**